### PR TITLE
Building a Tiny JS World

### PR DESCRIPTION
**Pull request by smillims**
### Building a Tiny JS World

[Demo](https://smillims.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/smillims/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Review by babayK0 — CHANGES_REQUESTED** [review]
Fine ! 
Lets do minimal requirements for this task: 

> Define objects representing this world inhabitants: a dog, a cat, a woman, and a man.

Also do some improvements:

Objects for species that do not have hands by natural design do not consequently have hands or any equivalent property

legs/hands property will not be phrase.

File: submissions/smillims/tiny-js-world/index.js
Comment (babayK0) [inline submissions/smillims/tiny-js-world/index.js:33]: `a dog, a cat, a woman, and a man.` 

Why don't you follow technical requirements?

**Review by babayK0 — CHANGES_REQUESTED** [review]

File: submissions/smillims/tiny-js-world/index.js
Comment (smillims) [reply submissions/smillims/tiny-js-world/index.js:33]: Wanted to did something unique.
Am I should correct it?

Comment (babayK0) [reply submissions/smillims/tiny-js-world/index.js:33]: @smillims maybe first do minimal requirements ?)

File: submissions/smillims/tiny-js-world/index.js
Comment (OleksiyRudenko) [inline submissions/smillims/tiny-js-world/index.js:46]: `Object` methods (`keys`, `values`, `entries`) as wel as `for...in` statement do not guarantee any particular order of keys/values. This means that under different JS engine the outcome might differ from expected.

Array guaranties the order. List properties or property names explicitly, map them into values (needed when list contains just names and not properties themselves) and then join.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@smillims nearly there!
A small fix and we're done.

File: submissions/smillims/tiny-js-world/index.js
Comment (OleksiyRudenko) [reply submissions/smillims/tiny-js-world/index.js:46]: Well, you will have to resolve the case when an object has no specific property from those explicitly listed.

`Array.filter` and `Array.includes` are to your rescue. (`Object.keys` is fine for filtering purposes since order doesn't matter when checking if a key is a part of a particular object's key list).

File: submissions/smillims/tiny-js-world/index.js
Comment (OleksiyRudenko) [inline submissions/smillims/tiny-js-world/index.js:53]: Please consider less generic names for functions and variables.
Check [this article](https://www.codewithjason.com/variable-name-anti-patterns/)

**Review by OleksiyRudenko — APPROVED** [review]
@smillims well done!

File: submissions/smillims/tiny-js-world/index.js
Comment (smillims) [reply submissions/smillims/tiny-js-world/index.js:53]: okay, I'm going to read it, thanks ^^

**Comment by smillims** [comment]
@babayK0 now waiting for your approve if everything okay ^^

File: submissions/smillims/tiny-js-world/index.js
Comment (babayK0) [inline submissions/smillims/tiny-js-world/index.js:49]: Naming
`returnString2` -> `returnString`

File: submissions/smillims/tiny-js-world/index.js
Comment (babayK0) [inline submissions/smillims/tiny-js-world/index.js:52]: Naming.
You iterate through array of objects, so `key`  is not good name for this variable.

```
const obj = { 
  a: 1
}

// a - key
// 1 - value
```
This is not a blocker, but further name your variables accordingly.

File: submissions/smillims/tiny-js-world/index.js
Comment (babayK0) [inline submissions/smillims/tiny-js-world/index.js:51]: You can do same result with [`Object.values`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values).
And you don't need `properties` array for this.

Also in [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Comparing_traditional_functions_to_arrow_functions) you can remove the body brackets and word "return"

**Review by babayK0 — CHANGES_REQUESTED** [review]
Watch my comments, after you fix i will approve. Good job !

**Review by babayK0 — APPROVED** [review]
LGTM!

**Pull request merged**
